### PR TITLE
SOC-1495 Make sure to check the edit token when creating new blog pages

### PR DESCRIPTION
**Pull request by garthwebb**
https://wikia-inc.atlassian.net/browse/SOC-1495

### CODE REVIEW

File: extensions/wikia/Blogs/monobook/SpecialCreateBlogPage.php [hunk 1]
```diff
@@ -7,71 +7,74 @@
...
-		if(function_exists('wfEditEnhancementsInit') && !class_exists('EditEnhancements')) {
-			wfEditEnhancementsInit(true);
+		if ( function_exists( 'wfEditEnhancementsInit' ) && !class_exists( 'EditEnhancements' ) ) {
+			wfEditEnhancementsInit( true );
 		}
 	}
 
 	public function execute() {
-		global $wgOut, $wgUser, $wgRequest, $wgTitle;
+		global $wgOut, $wgUser, $wgRequest;
```
Comment (garthwebb) [inline extensions/wikia/Blogs/monobook/SpecialCreateBlogPage.php:26]: I just reformatted the next two lines to span multiple lines.

File: extensions/wikia/Blogs/monobook/SpecialCreateBlogPage.php [hunk 2]
```diff
@@ -153,99 +155,102 @@ protected function save() {
...
-		$this->mFormData['postId'] = $wgRequest->getVal('blogPostId');
-		$this->mFormData['postTitle'] = $wgRequest->getVal('blogPostTitle');
-		$this->mFormData['postBody'] = $wgRequest->getVal('wpTextbox1');
-		$this->mFormData['postEditSummary'] = $wgRequest->getVal('wpSummary');
-		$this->mFormData['postCategories'] = $wgRequest->getVal('wpCategoryTextarea1');
-		$this->mFormData['isVotingEnabled'] = $wgRequest->getCheck('blogPostIsVotingEnabled');
-		$this->mFormData['isCommentingEnabled'] = $wgRequest->getCheck('blogPostIsCommentingEnabled');
-		$this->mFormData['isExistingArticleEditAllowed'] = $wgRequest->getVal('articleEditAllowed');
+		global $wgUser, $wgRequest;
+
```
Comment (garthwebb) [inline extensions/wikia/Blogs/monobook/SpecialCreateBlogPage.php:172]: These next 5 lines are the actual fix.  Look for the (already existing) token and make sure it matches the token for the current user.

**Comment by armonr** [comment]
:shipit:

File: extensions/wikia/Blogs/monobook/SpecialCreateBlogPage.php [hunk 3]
```diff
@@ -153,99 +155,102 @@ protected function save() {
...
+		$token = $wgRequest->getVal( 'wpEditToken' );
+		if ( !$wgUser->matchEditToken( $token ) ) {
+			$this->mFormErrors[] = wfMessage( 'create-blog-cant-edit' )->text();
```
Comment (kvas-damian) [inline extensions/wikia/Blogs/monobook/SpecialCreateBlogPage.php]: @garthwebb what do you think about more descriptive error message? Maybe `sessionfailure`?

BTW: message should be escaped.

Comment (garthwebb) [inline extensions/wikia/Blogs/monobook/SpecialCreateBlogPage.php]: Sounds good.  I should have just copied the line in the identical position in the non-monobook version of this controller.

**Comment by kvas-damian** [comment]
LGTM

**Pull request merged**